pinot-core/src/main/java/com/linkedin/pinot/core/util/FixedIntArrayOffHeapIdMap.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.util;
+
+import com.linkedin.pinot.common.Utils;
+import com.linkedin.pinot.core.io.readerwriter.PinotDataBufferMemoryManager;
+import com.linkedin.pinot.core.io.readerwriter.impl.FixedByteSingleValueMultiColumnReaderWriter;
+import com.linkedin.pinot.core.query.aggregation.function.customobject.FixedIntArray;
+import com.linkedin.pinot.core.realtime.impl.dictionary.BaseOffHeapMutableDictionary;
+import com.linkedin.pinot.core.segment.creator.impl.V1Constants;
+import java.io.IOException;
+import java.util.Arrays;
+import javax.annotation.Nonnull;
+
+
+/**
+ * Implementation of {@link IdMap} with {@link FixedIntArray} as key.
+ *
+ * This implementation also extends the {@link BaseOffHeapMutableDictionary} for code-reuse of off-heap functionality.
+ * However, it is not a full dictionary implementation (for example, does not implement getMin/Max etc).
+ *
+ */
+public class FixedIntArrayOffHeapIdMap extends BaseOffHeapMutableDictionary implements IdMap<FixedIntArray> {
```

Comment:
why is this a dictionary? Dictionary is more of BiMap whereas we just need a HashMap

Reply:
This is just extending BaseOffHeapMutableDictionary to re-use a lot of the off-heap functionality. Implementing the hashMap part of that requires taking the off-heap HashMap out of dictionary, which is a major change. Created issue https://github.com/linkedin/pinot/issues/2502 to address it in future.

Reply:
thanks.